src/main/java/cubicchunks/server/PlayerCubeMapEntry.java

```diff
@@ -147,7 +147,7 @@ public boolean sendToPlayers() {
 		if (this.sentToPlayers) {
 			return true;
 		}
-		if (this.cube == null || !this.cube.getGeneratorStage().isLastStage()) {
+		if (this.cube == null || !this.playerCubeMap.getWorld().getGeneratorPipeline().isLastStage(cube.getGeneratorStage())) {
```

Comment:
Or there would be a CLIENT "Stage", that would also be "final" stage, and would be private static class in the class that sets the stage for client.